src/main/java/com/google/opensesame/projects/ProjectDatastore.java

```diff
@@ -44,4 +58,56 @@ public static ProjectEntity getByRepositoryId(String repositoryId) {
 
     return ProjectEntity.fromProperties(datastoreEntity.getProperties());
   }
+
+  /**
+   * Gets a project entity by its repository ID. If it doesn't exist, creates a new ProjectEntity
+   * and returns that.
+   *
+   * @param repositoryId The GitHub repository ID.
+   * @return Returns an existing ProjectEntity or a new ProjectEntity with the specified repository
+   *     ID.
+   */
+  public static ProjectEntity getByRepositoryIdOrNew(String repositoryId) {
+    ProjectEntity projectEntity = getByRepositoryId(repositoryId);
+    if (projectEntity == null) {
+      projectEntity =
+          new ProjectEntity(repositoryId, new ArrayList<String>(), new ArrayList<String>());
+    }
+
+    return projectEntity;
+  }
```

Comment:
With this new function do we still need `getByRepositoryId`? It seems like it would be simpler to just return a new ProjectEntity in getByRepositoryId when it raises a`EntityNotFoundException`. However, maybe there is some benefit to returning Null when a project isn't found that I'm not thinking about.

Reply:
It's still sometimes going to be useful to know when a certain `ProjectEntity` was not found in the datastore. A good example would be when a user is making a get request to the `/project` servlet, where they must specify a specific `projectId`. In that case, if the `ProjectEntity` returned is null, the servlet knows to return an error response indicating to the user that the specified project does not exist.